### PR TITLE
Disable the embargo date elements until the embargo radio is selected

### LINKED ISSUE #316: Setting embargo dates should automatically select the "On this date" radio button
In the manage release section of the deposit form, when a user enters an embargo date, this should automatically select the "On this date" radio button. Currently I can leave "immediately" checked and choose a date, which makes it very unclear which option I'm selecting.

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
Fixes #316

### CODE REVIEW

File: app/components/works/available_date_component.html.erb [hunk 1]
```diff
@@ -1,31 +1,33 @@
-<div class="mb-3 row">
-  <div class="col-sm-10 release-option">
-    <%= form.radio_button :release, 'immediate', class: 'form-check-input' %>
-    <%= form.label :release_immediate, 'Immediately once deposit is completed' %>
+<div data-controller="complex-radio">
```
Comment (jcoyne) [inline app/components/works/available_date_component.html.erb:1]: Most of the changes in this file are indents to support the addition of this controller

File: app/components/works/available_date_component.html.erb [hunk 2]
```diff
@@ -1,31 +1,33 @@
...
+  <div class="mb-3 row">
+    <div class="col-sm-10 release-option" data-complex-radio-target="selection">
+      <%= form.radio_button :release, 'immediate', class: 'form-check-input', data: { action: 'complex-radio#disableUnselectedInputs' } %>
```
Comment (jcoyne) [inline app/components/works/available_date_component.html.erb:4]: These lines have substantive changes.

File: app/components/works/available_date_component.html.erb [hunk 3]
```diff
@@ -1,31 +1,33 @@
...
 
-<div class="embargo-date mb-5" data-edit-deposit-target="embargo-dateField">
-  <div class="row<%= ' is-invalid' if error? %>">
-    <div class="col-sm-10 release-option">
-      <%= form.radio_button :release, 'embargo', class: 'form-check-input' %>
-      <%= form.label :release_embargo, 'On this date' %>
+  <div class="embargo-date mb-5" data-edit-deposit-target="embargo-dateField">
+    <div class="row<%= ' is-invalid' if error? %>">
+      <div class="col-sm-10 release-option" data-complex-radio-target="selection">
+        <%= form.radio_button :release, 'embargo', class: 'form-check-input', data: { action: 'complex-radio#disableUnselectedInputs' } %>
```
Comment (jcoyne) [inline app/components/works/available_date_component.html.erb:12]: As do these.

File: app/components/works/available_date_component.html.erb (hunk 1 above)
Comment (mjgiarlo) [reply app/components/works/available_date_component.html.erb:1]: :beers: for making this clear.

**Review by mjgiarlo — COMMENTED** [review]
LGTM. Would you mind doing a couple before/after screenshots or a quick screencast to show it working? And/or consider adding a test to https://github.com/sul-dlss/happy-heron/blob/main/spec/features/single_radio_selection_spec.rb ?

**Comment by jcoyne** [comment]
@mjgiarlo I added a screenshot. I'm hesitant to add another feature spec as they are slow and you already tested this javascript in your PR.

**Review by mjgiarlo — APPROVED** [review]

**Comment by mjgiarlo** [comment]
@jcoyne I don't see the screenshot but NBD, I know what the other examples look like and you followed the same pattern here.

**Pull request merged**